clang/test/CheckedC/inferred-bounds/widened-bounds.c

```diff
@@ -464,3 +466,66 @@ void f19() {
 // CHECK:  [B1]
 // CHECK: upper_bound(p) = 3
 }
+
+void f20() {
+  // Declared bounds and deref offset are both INT_MAX. Valid widening.
+  _Nt_array_ptr<char> p : count(INT_MAX) = "";      // expected-error {{declared bounds for 'p' are invalid after initialization}}
+  if (*(p + INT_MAX))
+  {}
+
+// CHECK: In function: f20
+// CHECK:  [B12]
+// CHECK:    2: *(p + {{.*}})
+// CHECK:  [B11]
+// CHECK: upper_bound(p) = 1
+
+  // Declared bounds and deref offset are both INT_MIN. Valid widening.
+  _Nt_array_ptr<char> q : count(INT_MIN) = "";
+  if (*(q + INT_MIN))                               // expected-error {{out-of-bounds memory access}}
+  {}
+
+// CHECK:  [B10]
+// CHECK:    2: *(q + {{.*}})
+// CHECK:  [B9]
+// CHECK: upper_bound(q) = 1
+
+  // Declared bounds (INT_MIN) and deref offset (INT_MAX - 1). No sequential deref tests. No widening.
+  _Nt_array_ptr<char> r : count(INT_MIN) = "";
+  if (*(r + INT_MAX - 1))                               // expected-error {{out-of-bounds memory access}}
```

Comment:
Windows X86 Debug build fails to display the error `out-of-bounds memory access`. This seems to happen only at `*(p + INT_MAX)`. So for now, I have changed the dereference to `*(p + INT_MAX - 1)` to make this test pass. I have filed isue https://github.com/microsoft/checkedc-clang/issues/780. This needs to be investigated. I will come back and fix it later.

Reply:
Could you add a TODO before line 494 that includes the GitHub issue?